From c02689bc2861bc4ec71b62ae0e5d8e0849ec107d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 29 Apr 2021 19:57:20 +0300
Subject: [PATCH] [SW-0] fwupd: return general error on run failure

When the *_RUN command fails we return validation error.
This is not precise as this code does not really know the failure is
validation and not some other failure, because it does not parse any
status there.
Moreover, the VALID_ERR should be used for arguments and data validation
solely.
Change to return a general error.

Change-Id: I499dcbc6c67d32402b978fd15d2d2fc826a00be1
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 1dbb0fe..6e31ba5 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -231,7 +231,7 @@
 		memset((void *)base_offset, 0, ROUND_UP4(state.map_size));
 
 	if (err) {
-		sts = HL_UPD_STS_VALID_ERR;
+		sts = HL_UPD_STS_ERR;
 		preboot_puts("Failed to load FIT image\n");
 		goto fwupd_run_err;
 	}
